### PR TITLE
Replace ad hoc badges with Badge component

### PR DESCRIPTION
**Pull request by prasadkarmalkar**
## What?

In this PR replacing ad hoc badges with Badge component
Issue URL - https://github.com/WordPress/gutenberg/issues/68020
## How?

1. Exported Badge component.
2. Replaced span with Badge component in block editor list view, site editor and inspector
## Screenshots or screencast 

1. Site editor pages 
Screenshot 2024-12-18 at 3 55 05 PM

2. Inspector of editor
Screenshot 2024-12-18 at 3 55 59 PM

3. Editor list view anchor
Screenshot 2024-12-18 at 3 56 04 PM

### CODE REVIEW

**Comment by jameskoster** [comment]
Thanks for working on this ❤️ 

One issue is that the current anchor 'badge' that we find in List View truncates to avoid taking over he entire row. Currently the Badge component doesn't support truncation creating a bit of a problem. Here's a comparison:

| Trunk | This PR |
| --- | --- |
| Screenshot 2024-12-18 at 13 13 00 | Screenshot 2024-12-18 at 13 13 13 |

We could set a `max-width`, but the text will still overflow:

 
Screenshot 2024-12-18 at 13 16 02

My suggestion would be to tackle List View separately, after we've decided whether or not to add a `truncate` property to `Badge`.

File: packages/block-editor/src/components/list-view/block-select-button.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import clsx from 'clsx';
  * WordPress dependencies
  */
 import {
+	Badge,
```
Comment (tyxla) [inline packages/block-editor/src/components/list-view/block-select-button.js:10]: We currently don't expose `Badge` as a public component. You'll have to use the private API unlock mechanism to use it, see #68062 for a similar example.

**Review by tyxla — COMMENTED** [review]
This will currently throw errors as `Badge` import doesn't exist. See inline comment for more details.

**Comment by prasadkarmalkar** [comment]
Thanks everyone for the review. Since an alternative PR has been created for the issue, I will close this PR.

**Pull request closed without merging**